pkg/asset/installconfig/gcp/validation.go

```diff
@@ -50,12 +50,19 @@ func Validate(client API, ic *types.InstallConfig) error {
 		allErrs = append(allErrs, field.Invalid(field.NewPath("clusterName"), ic.ObjectMeta.Name, err.Error()))
 	}
 
+	sess, err := GetSession(context.TODO())
+	if err != nil {
+		allErrs = append(allErrs, field.InternalError(field.NewPath("session"), fmt.Errorf("failed to get gcp session")))
+	}
+	allErrs = append(allErrs, validateServiceAccountPresent(sess, ic)...)
```

Comment:
You should be able to follow the same pattern as the following functions, particularly I think `client` embeds a `Session` so you don't need to create one or handle errors here

Reply:
You can probably keep the function signature to just take a session but pass the one in the client rather than creating a new one